core/actions/actions.go

```diff
@@ -7,7 +7,7 @@ package actions
 import (
 	"strings"
 
-	"github.com/juju/charm/v13"
+	"github.com/juju/juju/charm"
```

Comment:
Core should not be importing from non core juju. There's a lot here, how are we going to handle it?

https://github.com/juju/juju/blob/79be87b162efa334f0388bdf905b9e460a91a5ad/core/doc.go#L21

Reply:
We could put it in core?

Reply:
That feels weird, but maybe that's what it needs. How would we integrate or not the current core/charm with the new charm package? There is already overlap, like the core/charm.origin containing a charm.channel.

Reply:
I think we should just put it into `core` in the long term. Short term I'm happy to put it into the root of juju as a temporary stop gap until we work out what terraform requires about charms. Once we know the exact usage patterns, we can design a real solution.

The fact that `core/charm` got around this problem just because it imported another juju module that should have been inside of juju/juju was a misstep. 

I think the pragmatic way forward is:

 1. Land this.
 2. Refactor `core/charm`
     - `core/charm/downloader` to internal/charm.
     - `core/charm/repository` to internal.
     - `core/charm/adapter.go` to internal.
     - `core/charm/channel.go` is just wrappers around the charm package.
     - move `FormatV2` from the charm package to `core/charm`.
     - `core/charm/format.go` should use a local interface and not the charm package.
     -  `code/charm/kubernetes.go` same as above.
 4. Formulate a way for terraform not to depend on this implicitly (either duplicate code or create a more focused library).